jellyfin-api/src/commonMain/kotlin/org/jellyfin/sdk/api/sockets/helper/ReconnectHelper.kt

```diff
@@ -1,9 +1,61 @@
 package org.jellyfin.sdk.api.sockets.helper
 
-internal class ReconnectHelper {
-	fun reportConnect() {}
-	fun reportDisconnect(){}
-	fun reset() {}
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import mu.KotlinLogging
+import kotlin.time.Duration.Companion.seconds
 
-	fun shouldReconnectNow() {}
+private val logger = KotlinLogging.logger {}
+
+internal class ReconnectHelper(
+	private val coroutineScope: CoroutineScope,
+	private val reconnect: suspend () -> Unit,
+) {
+	private var reconnectJob: Job? = null
+	private var attempts = 0
+
+	fun reset() {
+		logger.debug { "Resetting" }
+		reconnectJob?.cancel()
+		attempts = 0
+	}
+
+	fun notifyReconnect() {
+		reconnectJob?.cancel()
+		attempts++
+		logger.debug { "Notified about reconnect, attempts=${attempts}" }
+	}
+
+	fun notifyConnected() {
+		attempts = 0
+		logger.debug { "Notified about connect, attempts reset" }
+	}
+
+	fun scheduleReconnect(error: Boolean = false): Boolean {
+		reconnectJob?.cancel()
+
+		if (attempts > RETRY_ATTEMPTS) {
+			logger.debug { "Reconnect schedule failed: exceeded maximum retry attempts" }
+			return false
+		}
+
+		reconnectJob = coroutineScope.launch {
+			val retryAfter = if (error) RETRY_ERROR else RETRY_NORMAL
+
+			logger.info { "Reconnect scheduled in $retryAfter (error=$error)" }
+
+			delay(retryAfter)
+			reconnect()
```

Comment:
`reconnect` calls `notificyReconnect` which cancels the Job. Couldn't this cause issues?

Reply:
I didn't notice any issues with that while testing

Reply:
Alright. Might just be something to be aware of, in case there are any issues in the future.